src/components/Flags.tsx

```diff
@@ -1,72 +1,206 @@
-import { FC } from 'react';
+import { FC, useState } from 'react';
+import GolfIcon from '../assets/LegendIcons/golf_r.svg';
+import IndiaIcon from '../assets/LegendIcons/india_r.svg';
+import LimaIcon from '../assets/LegendIcons/lima_r.svg';
+import OscarIcon from '../assets/LegendIcons/oscar_r.svg';
+import RomeoIcon from '../assets/LegendIcons/romeo_r.svg';
+import ZuluIcon from '../assets/LegendIcons/zulu_r.svg';
 
 export const Flags: FC = () => {
+  const [description, setDescription] = useState<JSX.Element | null>(null);
+
   return (
-    <div className="my-12">
-      {data.map(({ letter, title, description }) => (
-        <Text key={letter} title={title} description={description} />
-      ))}
+    <div>
+      <div className="mapbox-legend-container">
+        {data.map(({ letter, title, description, icon }) => (
+          <LegendIcon
+            key={letter}
+            letter={letter}
+            title={title}
+            description={description}
+            icon={icon}
+            setDescription={setDescription}
+          />
+        ))}
+      </div>
+      <div className="mapbox-legend-item-description">
+        <p className="italic text-center">{description || ''}</p>
+      </div>
     </div>
   );
 };
 
-const Text: FC<{ title: string; description: string }> = ({
-  title,
-  description,
-}) => (
-  <div className="flex flex-col items-center">
-    <p className="font-serif">{title}</p>
-    <p className="text-muted italic text-center">{description}</p>
-  </div>
-);
+const LegendIcon: FC<{
+  letter: string;
+  title: string;
+  description: JSX.Element;
+  icon: FC;
+  setDescription: (description: JSX.Element | null) => void;
+}> = ({ letter, title, description, icon: Icon, setDescription }) => {
+  const [hovered, setHovered] = useState<boolean>(false);
+
+  return (
+    <div
+      className={`mapbox-legend-item ${hovered ? 'hovered' : ''}`}
```

Comment:
no need for this '`hovered`' class do we?

Reply:
Verdade ca dégage